pkg/platform/kube/platform.go

```diff
@@ -145,6 +145,22 @@ func (p *Platform) CreateFunction(createFunctionOptions *platform.CreateFunction
 		return nil, errors.Wrap(err, "Create function options validation failed")
 	}
 
+	// it's possible to pass a function without specifying any meta in the request, in that case skip getting existing function
+	if createFunctionOptions.FunctionConfig.Meta.Namespace != "" && createFunctionOptions.FunctionConfig.Meta.Name != "" {
+		existingFunctionConfig, err = p.getFunctionConfig(createFunctionOptions.FunctionConfig.Meta.Namespace,
+			createFunctionOptions.FunctionConfig.Meta.Name)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to get existing function config")
+		}
+	}
+
+	// if function exists, perform some validation with new function create options
```

Comment:
At this point it may not exist (`existingFunctionConfig = nil`)

Reply:
yeah, `if function exists`

Reply:
some of the validations check also if `existingFunctionConfig == nil` and `existingFunctionConfig != nil` and act correspondingly  
for the former: if function does not exists, its buildMode cannot be `neverBuild`
for the latter: validate resource version